checkov/arm/graph_builder/local_graph.py

```diff
@@ -29,12 +29,14 @@ def build_graph(self, render_variables: bool = False) -> None:
         self._create_vertices()
         logging.debug(f"[ArmLocalGraph] created {len(self.vertices)} vertices")
 
-        self._create_edges()
-        logging.debug(f"[ArmLocalGraph] created {len(self.edges)} edges")
+        self._create_vars_and_parameters_edges()
```

Comment:
shouldn't this be a part of the `create_edges` function from line 37?

Reply:
Yes, but the variable rendering needs to happen before the creation of the edges since some of the dependencies names can be brought from variables or parameters

Reply:
doesn't it work the other way around as well? meaning you might be missing variables to render if you didn't compute all edges first?

In terraform the order is "vertices->edges->render-vars", but I'm not sure this should also apply here, just to understand this is the correct order.

Also - if it is here specifically for that, it should probably be documented in the code as well that this part should be before variable renedering.